convert workflow status to varchar and index status column

MatthewBemis opened the pull request:
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Make sure liquibase is updated if appropriate
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell ![](http://i.imgur.com/9dLzbPd.png) that the PR exists if he wants to look at it **(apply requires_doge label)**
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] ![](http://i.imgur.com/9dLzbPd.png) sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

MatthewBemis commented on src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20160526_index_workflow_status.xml:
```diff
@@ -0,0 +1,16 @@
+
+    <changeSet logicalFilePath="dummy" author="mbemis" id="index-workflow-status-1">
+        <sql stripComments="true">
+            alter table WORKFLOW modify STATUS varchar(32);
```
workflow status was previously stored as TEXT, which is stored as a pointer to the value somewhere off the table. storing a small varchar instead allows us to nicely index statuses

MatthewBemis commented on src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20160526_index_workflow_status.xml:
```diff
@@ -0,0 +1,16 @@
+
+    <changeSet logicalFilePath="dummy" author="mbemis" id="index-workflow-status-2">
+        <sql stripComments="true">
+            alter table WORKFLOW add index (status);
```
now that we're avoiding lookups for status off of the table, we can index status allowing for queue status endpoint calls to avoid a full table scan and using all sorts of extra internal mysql functions (filesort, temporary tables)

dvoet commented:
👍

MatthewBemis commented:
@davidangb i believe you mentioned the other day that you were concerned about full table scans for the queue-status query, so this is probably of interest to you :)

davidangb commented:
Scope-creeping you ... could you also change submission.status, audit_submission_status.status, and audit_workflow_status.status?

jmthibault79 commented:
Would be good to make Slick aware of the index, and also specify the VARCHAR length as in #338

MatthewBemis commented:
@jmthibault79 i've now made slick aware of those things

@davidangb i updated the status type for those other columns. we use text instead of varchar in a lot of other places as well, so I added a ticket to the backlog for cleaning the rest of those up (GAWB-737)

jmthibault79 commented:
👍

davidangb commented:
👍 squash and go!

MatthewBemis committed:
stopped using text for statuses and added wf status index
